lib/memo_wise.rb

```diff
@@ -110,6 +110,13 @@ def allocate
         MemoWise::InternalAPI.create_memo_wise_state!(super)
       end
 
+      # This ensures that a memoized method defined on a parent class can still
+      # be used in a child class.
+      def inherited(subclass)
+        super
+        MemoWise::InternalAPI.create_memo_wise_state!(subclass)
+      end
```

Comment:
Should this be guarded in some way to make sure we need the MemoWise state on the class itself? I can't remember if we always create the class-level state, or only if we are memo_wise-ing class methods?

Reply:
Great question. I think ideally we'd only want to do this when we need it, but I don't see a particularly easy path to that right now, and I see the same issue with `def initialize` and `def allocate`. Would it make sense to make a separate issue for that work?

Reply:
Hmm I think they are different cases fundamentally.

- Initialize and allocate are **only** called on instances. They are not every triggered for classes, right?